c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,257 @@
+            .filter(Files::isRegularFile);
+    }
+
+    private List<Path> filterContent(List<Path> bundleFilePaths) {
+        List<Path> contentFilteredFilePaths = new ArrayList<>();
+        for (Path path : bundleFilePaths) {
+            String fileName = path.getFileName().toString();
+            try {
+                Path tempDirectory = createTempDirectory(null);
+                Path tempFile = Paths.get(tempDirectory.toAbsolutePath().toString(), fileName);
+                write(tempFile, (Iterable<String>) lines(path).filter(contentFilter)::iterator);
+                contentFilteredFilePaths.add(tempFile);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return contentFilteredFilePaths;
+    }
+
+    public Optional<byte[]> createDebugBundle(List<Path> filePaths) {
+        if (filePaths.size() == 0) {
+            return empty();
+        }
+        ByteArrayOutputStream byteOutputStream = null;
+        try {
+            byteOutputStream = new ByteArrayOutputStream();
+            try (GzipCompressorOutputStream gzipCompressorOutputStream = new GzipCompressorOutputStream(byteOutputStream);
+                 TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(gzipCompressorOutputStream)) {
+                for (Path filePath : filePaths) {
+                    TarArchiveEntry tarArchiveEntry = new TarArchiveEntry(filePath.toFile(), filePath.getFileName().toString());
+                    tarOutputStream.putArchiveEntry(tarArchiveEntry);
+                    copy(filePath, tarOutputStream);
+                    tarOutputStream.closeArchiveEntry();
+                }
+                tarOutputStream.finish();
+            }
+        } catch (Exception e) {
+            LOG.error("Error during create compressed bundle", e);
+            closeQuietly(byteOutputStream);
+            byteOutputStream = null;
+        } finally {
+            closeQuietly(byteOutputStream);
+        }
+        return ofNullable(byteOutputStream).map(ByteArrayOutputStream::toByteArray);
+    }
+
+    private void cleanup(List<Path> paths) {
+        paths.forEach(path -> {
+            try {
+                deleteIfExists(path);
+            } catch (IOException e) {
+                LOG.warn("Unable to delete temporary file", e);
+            }
+        });
+    }
+
+    interface DebugBundleFile {
+        String getFileName();
+    }
+
+    enum LogFile implements DebugBundleFile {
+        APP_LOG_FILE("minifi-app.log"),
+        BOOTSTRAP_LOG_FILE("minifi-bootstrap.log");
+
+        private final String fileName;
+
+        LogFile(String fileName) {
+            this.fileName = fileName;
+        }
+
+        public String getFileName() {
+            return fileName;
+        }
+    }
+
+    enum ConfigFile implements DebugBundleFile {
+        CONFIG_YML_FILE("config.yml"),
+        CONFIG_NEW_YML_FILE("config-new.yml"),
```

Comment:
I think we don't need to add the config-new.yml to the debug bundle. It's basically a temporary file that is used to construct the config.yml during config updates.

Reply:
Thanks, excluded it from the list of the files